pkg/validations/base_test.go

```diff
@@ -202,6 +203,79 @@ func TestIncompatibleChecksAreDisabled(t *testing.T) {
 	}
 }
 
+// Test to check if a resource has 0 replicas it clears metrics
+func TestValidateZeroReplicas(t *testing.T) {
+
+	customCheck := newCustomCheck()
+
+	// Initialize Engine
+	err := initializeEngine(customCheck)
+	if err != nil {
+		t.Errorf("Error initializing engine %v", err)
+	}
+
+	request := reconcile.Request{
+		NamespacedName: types.NamespacedName{Name: "foo", Namespace: "bar"},
+	}
+
+	// Setup test deployment file with 0 replicas
+	replicaCnt := int32(0)
+	deployment, err := createTestDeployment(replicaCnt)
+	if err != nil {
+		t.Errorf("Error creating deployment from template %v", err)
+	}
+
+	// Run validations against test environment
+	RunValidations(request, deployment, testutils.ObjectKind(deployment), false)
+
+	// Acquire labels generated from validations
+	labels := getPromLabels(request.Namespace, request.Name, "Deployment")
+
+	// The 'GetMetricWith()' function will create a new metric with provided labels if it
+	// does not exist. The default value of a metric is 0. Therefore, a value of 0 implies we
+	// successfully cleared the metric label combination.
+	metric, err := engine.GetMetric(customCheck.Name).GetMetricWith(labels)
+	if err != nil {
+		t.Errorf("Error getting prometheus metric: %v", err)
+	}
+
+	// If metrics are cleared then the value will be == 0
+	if metricValue := int(prom_tu.ToFloat64(metric)); metricValue != 0 {
+		t.Errorf("Deployment test failed %#v: got %d want %d", customCheck.Name, metricValue, 0)
+	}
+
+	// Check using a replica count above 0
+	replicaCnt = int32(3)
```

Comment:
Setting 3 replicas will cause the check to pass and no metric would be generated.  I would use 1 here so the check will fail and create a metric.